app-modules/ai/tests/Feature/Http/Controllers/CompleteResponseControllerTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use App\Models\User;
+use Mockery\MockInterface;
+
+use function Pest\Laravel\post;
+use function Tests\asSuperAdmin;
+
+use AdvisingApp\Ai\Enums\AiModel;
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Ai\Models\AiMessage;
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Enums\AiApplication;
+use AdvisingApp\Ai\Actions\CompleteResponse;
+use AdvisingApp\Ai\Exceptions\AiThreadLockedException;
+use AdvisingApp\Ai\Exceptions\AiAssistantArchivedException;
+
+it('completes the last response in a thread', function () {
+    asSuperAdmin();
+
+    $responseContent = AiMessage::factory()->make()->content;
+
+    /** @phpstan-ignore-next-line */
+    $this->mock(
```

Comment:
Isn't there a Pest `mock` function you could import and use here so you don't have to use `$this` and ignore the phpstan issue?

Reply:
I have tried, and there doesn't seem to be. `$this->mock()` on the TestCase actually binds the mocked instance to the Laravel container

```php
$this->instance($abstract, Mockery::mock(...array_filter(func_get_args())))
```

Whereas PEST's integration seems to be framework agnostic and not bind